`DoubleDeserializer` rejects "+INF"/"+Infinity" despite accepting "INF"/"Infinity" and "-INF"/"-Infinity"

### Search before asking

- [x] I searched in the [issues](https://github.com/FasterXML/jackson-databind/issues) and found nothing similar.

### Describe the bug

**StdDeserializer._checkDoubleSpecialValue()** accepts "INF" and "-INF" as special double values but rejects "+INF", throwing InvalidFormatException. The "+INF" form is the explicitly-signed positive infinity representation defined in
  https://www.w3.org/TR/xmlschema11-2/#double and is commonly emitted by C++ XML serializers.

 ### Current behavior                                                                                 

  | Input | Result |                                                                                                                                                                                                                                
  |---|---|
  | `"INF"` | `Double.POSITIVE_INFINITY` |                                                                                                                                                                                                          
  | `"Infinity"` | `Double.POSITIVE_INFINITY` |                                                        
  | `"-INF"` | `Double.NEGATIVE_INFINITY` |                                                                                                                                                                                                         
  | `"-Infinity"` | `Double.NEGATIVE_INFINITY` |
  | `"NaN"` | `Double.NaN` |                                                                                                                                                                                                                        
  | `"+INF"` | `InvalidFormatException` |                                                              
  | `"+Infinity"` | `InvalidFormatException` |                                                                                                                                                                                                      
                                                                                                                                                                                                                                                    

### Version Information

issue reproduced using 2.20.1 

still the same code in the latest commit

### Reproduction

Can be easily reproduced by 
1. creating an xml string containing double type with "+INF"
2. encode into bytes
3. use ObjectMapper to readValue()


Stack Trace (some sensitive parts are omitted)
```
Caused by: com.fasterxml.jackson.databind.exc.InvalidFormatException: Cannot deserialize value of type `java.lang.Double` from String "+INF": not a valid `Double` value
 at [...omitted...])
	at com.fasterxml.jackson.databind.exc.InvalidFormatException.from(InvalidFormatException.java:67)
	at com.fasterxml.jackson.databind.DeserializationContext.weirdStringException(DeserializationContext.java:2014)
	at com.fasterxml.jackson.databind.DeserializationContext.handleWeirdStringValue(DeserializationContext.java:1300)
	at com.fasterxml.jackson.databind.deser.std.NumberDeserializers$DoubleDeserializer._parseDouble(NumberDeserializers.java:785)
	at com.fasterxml.jackson.databind.deser.std.NumberDeserializers$DoubleDeserializer.deserialize(NumberDeserializers.java:704)
	at com.fasterxml.jackson.databind.deser.std.NumberDeserializers$DoubleDeserializer.deserialize(NumberDeserializers.java:683)
	at com.fasterxml.jackson.databind.deser.impl.FieldProperty.deserializeAndSet(FieldProperty.java:137)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:302)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:169)
	at com.fasterxml.jackson.databind.deser.impl.FieldProperty.deserializeAndSet(FieldProperty.java:137)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:302)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:169)
	at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer._deserializeNoNullChecks(CollectionDeserializer.java:501)
	at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer._deserializeFromArray(CollectionDeserializer.java:358)
	at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer.deserialize(CollectionDeserializer.java:245)
	at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer.deserialize(CollectionDeserializer.java:29)
	at com.fasterxml.jackson.databind.deser.impl.FieldProperty.deserializeAndSet(FieldProperty.java:137)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:302)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:169)
	at com.fasterxml.jackson.dataformat.xml.deser.WrapperHandlingDeserializer.deserialize(WrapperHandlingDeserializer.java:122)
	at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer._deserializeNoNullChecks(CollectionDeserializer.java:501)
	at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer._deserializeFromArray(CollectionDeserializer.java:358)
	at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer.deserialize(CollectionDeserializer.java:245)
	at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer.deserialize(CollectionDeserializer.java:29)
	at com.fasterxml.jackson.databind.deser.impl.FieldProperty.deserializeAndSet(FieldProperty.java:137)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:302)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:169)
	at com.fasterxml.jackson.dataformat.xml.deser.WrapperHandlingDeserializer.deserialize(WrapperHandlingDeserializer.java:122)
	at com.fasterxml.jackson.databind.deser.impl.FieldProperty.deserializeAndSet(FieldProperty.java:137)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:302)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:169)
	at com.fasterxml.jackson.dataformat.xml.deser.XmlDeserializationContext.readRootValue(XmlDeserializationContext.java:104)
	at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:5038)
	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3991)
```

### Expected behavior

 ### Expected behavior
                                                                                                                                                                                                                                                    
  `"+INF"` and `"+Infinity"` should deserialize to `Double.POSITIVE_INFINITY`, consistent with how `"-INF"` and `"-Infinity"` deserialize to `Double.NEGATIVE_INFINITY`. 

### Additional context

 Is this a valid concern? I am reporting this as my company uses this amazing library and encountered this exception

--- a/release-notes/CREDITS-2.x
+++ b/release-notes/CREDITS-2.x
@@ -2023,6 +2023,6 @@ Alexey Tsvetkov (@AlexeyTsvetkov)
   [2.21.3]
 
 Kenta Yoshii (@KentaYoshii)
- * Reported #5898: `DoubleDeserializer` rejects "+INF"/"+Infinite" despite accepting
-   "INF"/"Infinite" and "-INF"/"-Infinite"
+ * Reported #5898: `DoubleDeserializer` rejects "+INF"/"+Infinity" despite accepting
+   "INF"/"Infinity" and "-INF"/"-Infinity"
   [2.21.3]

--- a/release-notes/CREDITS-2.x
+++ b/release-notes/CREDITS-2.x
@@ -2021,3 +2021,8 @@ Alexey Tsvetkov (@AlexeyTsvetkov)
  * Reported #5840: Jackson 2.21 throws Conflicting property-based creators if both
    default (0-arg) and multi-arg constructor annotated
   [2.21.3]
+
+Kenta Yoshii (@KentaYoshii)
+ * Reported #5898: `DoubleDeserializer` rejects "+INF"/"+Infinite" despite accepting
+   "INF"/"Infinite" and "-INF"/"-Infinite"
+  [2.21.3]

--- a/release-notes/VERSION-2.x
+++ b/release-notes/VERSION-2.x
@@ -15,6 +15,10 @@ Project: jackson-databind
 #5891: `EnumMapDeserializer._deserializeUsingProperties()` corrupts parser
   state after skipping unknown enum keys
  (fixed by Lee Jiwon)
+#5898: `DoubleDeserializer` rejects "+INF"/"+Infinite" despite accepting
+  "INF"/"Infinite" and "-INF"/"-Infinite"
+ (reported by Kenta Y)
+ (fix by @cowtowncoder)
 
 2.21.2 (20-Mar-2026)
 

--- a/release-notes/VERSION-2.x
+++ b/release-notes/VERSION-2.x
@@ -15,8 +15,8 @@ Project: jackson-databind
 #5891: `EnumMapDeserializer._deserializeUsingProperties()` corrupts parser
   state after skipping unknown enum keys
  (fixed by Lee Jiwon)
-#5898: `DoubleDeserializer` rejects "+INF"/"+Infinite" despite accepting
-  "INF"/"Infinite" and "-INF"/"-Infinite"
+#5898: `DoubleDeserializer` rejects "+INF"/"+Infinity" despite accepting
+  "INF"/"Infinity" and "-INF"/"-Infinity"
  (reported by Kenta Y)
  (fix by @cowtowncoder)
 

--- a/src/main/java/com/fasterxml/jackson/databind/deser/std/StdDeserializer.java
+++ b/src/main/java/com/fasterxml/jackson/databind/deser/std/StdDeserializer.java
@@ -284,7 +284,7 @@ public abstract class StdDeserializer<T>
             if (inst.canCreateFromInt()) {
                 if (ctxt.findCoercionAction(LogicalType.Integer, Integer.class,
                         CoercionInputShape.String) == CoercionAction.TryConvert) {
-                    return (T) inst.createFromInt(ctxt, _parseIntPrimitive(p, ctxt, value));
+                    return (T) inst.createFromInt(ctxt, _parseIntPrimitive(ctxt, value));
                 }
             }
             if (inst.canCreateFromLong()) {
@@ -1162,7 +1162,7 @@ public abstract class StdDeserializer<T>
         }
 
         final CoercionAction act = _checkFromStringCoercion(ctxt, text,
-                LogicalType.Float, Float.TYPE);
+                LogicalType.Integer, Float.TYPE);
         if (act == CoercionAction.AsNull) {
             // 03-May-2021, tatu: Might not be allowed (should we do "empty" check?)
             _verifyNullForPrimitive(ctxt);
@@ -1310,7 +1310,7 @@ public abstract class StdDeserializer<T>
         }
 
         final CoercionAction act = _checkFromStringCoercion(ctxt, text,
-                LogicalType.Float, Double.TYPE);
+                LogicalType.Integer, Double.TYPE);
         if (act == CoercionAction.AsNull) {
             // 03-May-2021, tatu: Might not be allowed (should we do "empty" check?)
             _verifyNullForPrimitive(ctxt);

--- a/src/main/java/com/fasterxml/jackson/databind/deser/std/StdDeserializer.java
+++ b/src/main/java/com/fasterxml/jackson/databind/deser/std/StdDeserializer.java
@@ -1235,6 +1235,8 @@ public abstract class StdDeserializer<T>
         if (!text.isEmpty()) {
             switch (text.charAt(0)) {
             case 'I':
+            // 11-Apr-2026, [databind#5898]: accept "+Infinity", "+INF"
+            case '+':
                 if (_isPosInf(text)) {
                     return Float.POSITIVE_INFINITY;
                 }
@@ -1393,6 +1395,8 @@ public abstract class StdDeserializer<T>
         if (!text.isEmpty()) {
             switch (text.charAt(0)) {
             case 'I':
+            // 11-Apr-2026, [databind#5898]: accept "+Infinity", "+INF"
+            case '+':
                 if (_isPosInf(text)) {
                     return Double.POSITIVE_INFINITY;
                 }

--- a/src/main/java/com/fasterxml/jackson/databind/deser/std/StdDeserializer.java
+++ b/src/main/java/com/fasterxml/jackson/databind/deser/std/StdDeserializer.java
@@ -284,7 +284,7 @@ public abstract class StdDeserializer<T>
             if (inst.canCreateFromInt()) {
                 if (ctxt.findCoercionAction(LogicalType.Integer, Integer.class,
                         CoercionInputShape.String) == CoercionAction.TryConvert) {
-                    return (T) inst.createFromInt(ctxt, _parseIntPrimitive(ctxt, value));
+                    return (T) inst.createFromInt(ctxt, _parseIntPrimitive(p, ctxt, value));
                 }
             }
             if (inst.canCreateFromLong()) {
@@ -1162,7 +1162,7 @@ public abstract class StdDeserializer<T>
         }
 
         final CoercionAction act = _checkFromStringCoercion(ctxt, text,
-                LogicalType.Integer, Float.TYPE);
+                LogicalType.Float, Float.TYPE);
         if (act == CoercionAction.AsNull) {
             // 03-May-2021, tatu: Might not be allowed (should we do "empty" check?)
             _verifyNullForPrimitive(ctxt);
@@ -1308,7 +1308,7 @@ public abstract class StdDeserializer<T>
         }
 
         final CoercionAction act = _checkFromStringCoercion(ctxt, text,
-                LogicalType.Integer, Double.TYPE);
+                LogicalType.Float, Double.TYPE);
         if (act == CoercionAction.AsNull) {
             // 03-May-2021, tatu: Might not be allowed (should we do "empty" check?)
             _verifyNullForPrimitive(ctxt);
@@ -1628,7 +1628,9 @@ public abstract class StdDeserializer<T>
     }
 
     protected final boolean _isPosInf(String text) {
-        return "Infinity".equals(text) || "INF".equals(text);
+        return "Infinity".equals(text) || "INF".equals(text)
+                // 11-Apr-2026, [databind#5898]: also accept:
+                || "+Infinity".equals(text) || "+INF".equals(text);
     }
 
     protected final boolean _isNaN(String text) { return "NaN".equals(text); }

--- a/src/test/java/com/fasterxml/jackson/databind/deser/jdk/JDKNumberDeserTest.java
+++ b/src/test/java/com/fasterxml/jackson/databind/deser/jdk/JDKNumberDeserTest.java
@@ -230,6 +230,26 @@ public class JDKNumberDeserTest
         assertEquals(Double.valueOf(Double.NEGATIVE_INFINITY), result);
     }
 
+    // [databind#5898]: accept "+INF" and "+Infinity" as positive infinity
+    @Test
+    public void testDoublePlusInf5898() throws Exception
+    {
+        assertEquals(Double.POSITIVE_INFINITY, MAPPER.readValue(q("+Infinity"), Double.class));
+        assertEquals(Double.POSITIVE_INFINITY, MAPPER.readValue(q("+Infinity"), Double.TYPE));
+        assertEquals(Double.POSITIVE_INFINITY, MAPPER.readValue(q("+INF"), Double.class));
+        assertEquals(Double.POSITIVE_INFINITY, MAPPER.readValue(q("+INF"), Double.TYPE));
+    }
+
+    // [databind#5898]: accept "+INF" and "+Infinity" as positive infinity
+    @Test
+    public void testFloatPlusInf5898() throws Exception
+    {
+        assertEquals(Float.POSITIVE_INFINITY, MAPPER.readValue(q("+Infinity"), Float.class));
+        assertEquals(Float.POSITIVE_INFINITY, MAPPER.readValue(q("+Infinity"), Float.TYPE));
+        assertEquals(Float.POSITIVE_INFINITY, MAPPER.readValue(q("+INF"), Float.class));
+        assertEquals(Float.POSITIVE_INFINITY, MAPPER.readValue(q("+INF"), Float.TYPE));
+    }
+
     // 01-Mar-2017, tatu: This is bit tricky... in some ways, mapping to "empty value"
     //    would be best; but due to legacy reasons becomes `null` at this point
     @Test

